nixos/modules/security/default.nix

```diff
@@ -0,0 +1,32 @@
+{ config, lib, ... }:
+let
+  cfg = config.security;
+in
+{
+  options = {
+    security.lsm = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
+      description = ''
+        A list of the LSMs to initialize in order.
+      '';
+    };
+  };
+
+  config = lib.mkIf (lib.lists.length cfg.lsm > 0) {
+    assertions = [
+      {
+        assertion =
+          let
+            keys = lib.map (v: builtins.elemAt (lib.splitString "=" v) 0) config.boot.kernelParams;
```

Comment:
I think the conflict @notashelf was referring to was duplicate things in the `lsm` part of the config, so you can just unique cfg.lsm. That being said, we should probably _also_ error if they provide security= anywhere regardless of what it is, if lsm is enabled.

Reply:
Ok, I've made the option use `mergeUniqueOption` so it errors on duplicates.

Reply:
Ok, apparently `merge` doesn't exist with `lib.mkOption`. I've switched to the `uniq` type. I've also added an assert to fail when `security=` is used.

Reply:
Do we need this first assertion then?

Reply:
Probably not, I've removed it. Things should be good now.